src/wayland/shell/kde/decoration.rs

```diff
@@ -0,0 +1,129 @@
+//! KDE Window decoration manager
+//!
+//! This interface allows a compositor to announce support for KDE's legacy server-side decorations.
+//!
+//! A client can use this protocol to request being decorated by a supporting compositor.
+//!
+//! ```
+//! extern crate wayland_server;
+//! extern crate smithay;
+//!
+//! use smithay::delegate_kde_decoration;
+//! use smithay::wayland::shell::kde::decoration::{KdeDecorationHandler, KdeDecorationState};
+//!
+//! # struct State { kde_decoration_state: KdeDecorationState };
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//!
+//! // Create the new KdeDecorationState.
+//! let state = KdeDecorationState::new::<State, _>(&display.handle(), None);
+//!
+//! // Insert KdeDecorationState into your compositor state.
+//! // …
+//!
+//! // Implement KDE server decoration handlers.
+//! impl KdeDecorationHandler for State {
+//!     fn kde_decoration_state(&self) -> &KdeDecorationState {
+//!         &self.kde_decoration_state
+//!     }
+//! }
+//!
+//! delegate_kde_decoration!(State);
+//! ```
+
+use slog::Logger;
+use wayland_protocols_misc::server_decoration::server::org_kde_kwin_server_decoration::{
+    Mode, OrgKdeKwinServerDecoration,
+};
+use wayland_protocols_misc::server_decoration::server::org_kde_kwin_server_decoration_manager::OrgKdeKwinServerDecorationManager;
+use wayland_server::backend::GlobalId;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::{Dispatch, DisplayHandle, GlobalDispatch, WEnum};
+
+/// KDE server decoration handler.
+pub trait KdeDecorationHandler {
+    /// Return the KDE server decoration state.
+    fn kde_decoration_state(&self) -> &KdeDecorationState;
+
+    /// Handle decoration manager creation.
+    ///
+    /// Allows setting up the decoration manager, like setting the default decoration mode using
+    /// [`OrgKdeKwinServerDecorationManager::default_mode`].
+    fn setup(&mut self, _kde_decoration_manager: &OrgKdeKwinServerDecorationManager) {}
+
+    /// Handle new decoration object creation.
+    ///
+    /// Called whenever a new decoration object is created, usually this happens when a new window
+    /// is opened.
+    fn new_decoration(&mut self, _surface: &WlSurface, _decoration: &OrgKdeKwinServerDecoration) {}
+
+    /// Handle surface decoration mode requests.
+    ///
+    /// Called when a surface requests a specific decoration mode or acknowledged the compositor's
+    /// decoration request.
+    ///
+    /// **It is up to the compositor to prevent feedback loops**, a client is free to ignore modes
+    /// suggested by [`OrgKdeKwinServerDecoration::mode`] and instead keep requesting their
+    /// preferred mode instead.
+    fn request_mode(
+        &mut self,
+        _surface: &WlSurface,
+        _decoration: &OrgKdeKwinServerDecoration,
+        _mode: WEnum<Mode>,
+    ) {
+    }
+
+    /// Handle decoration object removal for a surface.
+    fn release(&mut self, _surface: &WlSurface) {}
```

Comment:
Is it possible to provide the handle here?

Reply:
What handle?

Reply:
The kde decoration protocol object